integration/nwo/deploy.go

```diff
@@ -506,6 +514,69 @@ func checkCommitReadiness(n *Network, peer *Peer, channel string, chaincode Chai
 	}
 }
 
+// ApproveIncorrectChaincodeForEachOrg approves a deliberately erroneous chaincode definition for each organization,
+// to test the network's ability to detect and handle discrepancies in chaincode definitions across different organizations.
+func ApproveIncorrectChaincodeForEachOrg(n *Network, channel string, orderer *Orderer, chaincode Chaincode, peers ...*Peer) {
+	// used to ensure we only approve once per org
+	approvedOrgs := map[string]bool{}
+	for _, p := range peers {
+		if _, ok := approvedOrgs[p.Organization]; !ok {
+			sess, err := n.PeerAdminSession(p, commands.ChaincodeApproveForMyOrg{
+				ChannelID:           channel,
+				Orderer:             n.OrdererAddress(orderer, ListenPort),
+				Name:                chaincode.Name,
+				Version:             "incorrect-version", // Intentionally set incorrect version
```

Comment:
Much improved, the test now shows the intent of the new inspect option.

I still have a question though... if all the orgs approve the same incorrect version, won't there be no discrepancies to detect? Maybe I'm missing something...

Reply:
@denyeart In this case, I assume that the organization executing commitreadiness has also specified the incorrect parameters in its own approve  mistakenly.
Since the correct chaincode definition is specified in the argument to commitreadiness, all organizations will detect the mismatch.

Other scenario ideas are a case where the organization executing commitreadiness with inspect only specifies the correct version, or where only one organization specifies incorrect version. In these scenarios, the test code may become a bit more complex.

Reply:
Ah right, they all approve the same incorrect definition, but checkcommitreadiness checks for the correct definition. Looks good!